services/gateway/service/auth.go

```diff
@@ -0,0 +1,59 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Drozd0f/ttto-go/gen/proto/auth"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	ErrValidation = errors.New("validation error")
+	ErrUserAlreadyExists   = errors.New("user already exist")
+	ErrUserNotExists     = errors.New("user not exist")
+)
+
+func (s *Service) CreateUser(ctx context.Context, cur *auth.CreateUserRequest) error {
+	_, err := s.ac.CreateUser(ctx, cur)
+	if err != nil {
+		stat, ok := status.FromError(err)
+		if ok {
+			switch {
+			case stat.Code() == codes.InvalidArgument:
+				return ErrValidation
+			case stat.Code() == codes.AlreadyExists:
+				return ErrUserAlreadyExists
+			}
+			return fmt.Errorf("auth service create user: %s", stat.Message())
+		}
+
+		return fmt.Errorf("auth service status from error: %w", ErrGrpcBadError)
```

Comment:
Not fully understand why you create this error, it unknown error from other service, that will be returned as "Internal error" and logged